### DISCUSSION TITLE
Use wagmi without connectors and perform transactions

### DISCUSSION THREAD

**Discussion by msindal** [wevm/wagmi#4295] [Q&A]
Is there a way for me to perform transactions without passing connectors array. I am actually developing an sdk on top of wagmi/core and I configured it with basic configuration without passing connectors. But while executing writeContract function I got this error: main-m_cOqYyv.js:5037 Uncaught (in promise) ConnectorNotConnectedError

Now I am connected to metamask but haven't passed metamask connector in connectors array. Can I not pass wallets and execute?

**Comment by tmm** [comment]
What does your code look like? Depending on what you are trying to do [Viem](https://viem.sh) by itself might be a better fit.
